### ISSUE TITLE
dd-java-agent:0.68.0 with spring-boot 2.4.0 doesn't send 'netty.request' traces to datadog apm

### ISSUE THREAD

**Issue by young-druid** [DataDog/dd-trace-java#2117]
I created a sample project [here](https://github.com/young-druid/dd-agent-apm-test). I hope you can clone it.
Please, do the following steps:

- `./gradlew clean bootRun`. Wait until application is strated.
- execute `curl -X GET http://localhost:8080/test `
- stop spring application with Ctrl+C 
- check this `build/dd-agent/log.txt` file. You won't find that any traces for `netty.request`.

After that, downgrade `id 'org.springframework.boot' version '2.4.0'` to `id 'org.springframework.boot' version '2.3.5.RELEASE'` in `build.gradle`. Repeat the steps above. If you check log.txt you can find that traces for `netty.request` are being sent.

It looks like versions of spring-web-flux, netty and projectreactor which can be found in the latest spring-boot 2.4.0 should be supported by dd-java-agent. But for some reason in the latest spring-boot you cannot get `netty.request` traces in datadog apm because of this bug (if it is a bug and not some configuration issue).

**Comment by dkulig** [comment]
Thumbs up for this one. In my case, the B3 Headers extraction no longer works after upgrading to SpringBoot 2.4.0. Looks like its related.

**Comment by young-druid** [comment]
I updated test project a bit:
```
1. Updated Kotlin version to 1.4.21
2. Updated Spring boot version to 2.4.1
3. Updated datadog agent to 0.69.0
```
I can still see that `netty.request` is not being sent for spring-boot 2.4.1 while it is nicely sent for spring-boot 2.3.5.RELEASE

**Comment by devinsba** [comment]
Thanks for the report. We will need to look into this further. Off the top of my head, this is the first version of boot+webflux that includes `reactor-netty` 1.x so we likely need to check here for breaking changes that we need to handle

**Comment by jorgebo10** [comment]
Hi, in my case neither dd.trace_id nor dd.span_id are been injected in logs. Using Spring boot >= 2.4.0. Had to downgrade to 2.3.8.RELEASE

**Comment by devinsba** [comment]
This is being actively investigated. Please keep an eye out, I'll drop a comment once a PR to fix this is created

**Comment by devinsba** [comment]
@young-druid I tested your project with version `0.75.0` of dd-java-agent and I see a `netty-request` span now. I suspect this fixed the issue https://github.com/DataDog/dd-trace-java/pull/2415. I'll keep digging to make sure I'm confident saying this issue is fixed but you can probably try the latest tracer to see if that works for you

<img width="969" alt="Screen Shot 2021-03-16 at 10 47 30 AM" src="https://user-images.githubusercontent.com/139938/111329027-29f94700-8645-11eb-86e8-365d86044be9.png">

**Comment by devinsba** [comment]
I can't find any evidence that this wasn't fixed in release `0.75.0`. If someone finds a configuration that doesn't work please create a new issue, a fork of the project above that recreates would be great too.

**Comment by workmanw** [comment]
We can also confirm that this issue is fixed for us in `0.75.0` (using Spring Cloud Gateway). Thanks a lot!!

**Comment by young-druid** [comment]
It works with the latest Spring Boot 2.4.3. Thanks.
